cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java

```diff
@@ -227,16 +237,64 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
+        for (Map.Entry<String, String> defaultEntry : userConfiguration.getCruiseControlDefaultPropertiesMap().entrySet()) {
+            if (userConfiguration.getConfigOption(defaultEntry.getKey()) == null) {
+                userConfiguration.setConfigOption(defaultEntry.getKey(), defaultEntry.getValue());
             }
         }
-        cruiseControl.setConfiguration(configuration);
+        // Ensure that the configured anomaly.detection.goals are a sub-set of the default goals
+        checkGoals(userConfiguration);
+        cruiseControl.setConfiguration(userConfiguration);
         return cruiseControl;
     }
 
+    /**
+     *  This method ensures that the checks in cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java
+     *  sanityCheckGoalNames() method (L118)  don't fail if a user submits custom default goals that have less members then the default
+     *  anomaly.detection.goals.
+     * @param configuration The configuration instance to be checked.
+     * @throws UnsupportedOperationException If the configuration contains self.healing.goals configurations.
+     */
+    public static void checkGoals(CruiseControlConfiguration configuration) {
+        // If self healing goals are defined then these take precedence
+        if (configuration.getConfigOption(CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY) != null) {
+            String selfHealingGoalsString = configuration.getConfigOption(CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY);
+            List<String> selfHealingGoals = Arrays.asList(selfHealingGoalsString.split("\\s*,\\s*"));
+            if (!selfHealingGoals.isEmpty()) {
+                throw new UnsupportedOperationException("Cruise Control's self healing functionality is not currently supported. Please remove " +
+                        CRUISE_CONTROL_SELF_HEALING_CONFIG_KEY + " config");
+            }
+        }
+
+        // Either self.healing.goals is null or it is an empty list
+        String anomalyGoalsString;
+        if (configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY) != null) {
+            // The user has defined some anomaly detection goals
+            anomalyGoalsString = configuration.getConfigOption(CRUISE_CONTROL_ANOMALY_DETECTION_CONFIG_KEY);
+        } else {
+            // No anomaly detection goals have been defined by the user, therefore the defaults defined in Cruise Control
+            // will be used.
+            anomalyGoalsString = CRUISE_CONTROL_DEFAULT_ANOMALY_DETECTION_GOALS;
+        }
+
+        Set<String> anomalyDetectionGoals = new HashSet<>(Arrays.asList(anomalyGoalsString.split("\\s*,\\s*")));
+
+        String defaultGoalsString = configuration.getConfigOption(CRUISE_CONTROL_DEFAULT_GOALS_CONFIG_KEY);
+        Set<String> defaultGoals = new HashSet<>(Arrays.asList(defaultGoalsString.split("\\s*,\\s*")));
```

Comment:
We set it to this [1] here [2] if default.goals is set to null


[1] https://github.com/strimzi/strimzi-kafka-operator/blob/master/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControlConfiguration.java#L26-L41
[2] https://github.com/strimzi/strimzi-kafka-operator/blob/e1cad47d1a4f70a812e14895e8d1eb09a3964095/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java#L242-L244

Reply:
Which is great and true _today_, but there's no harm in adding a null check in case it's not true _tomorrow_. Remember, the more distant the place where you know the value is always set then the more fragile it is to a future change violating the non-nullness.